src/common/CMakeLists.txt

```diff
@@ -1,10 +1,13 @@
 # Compiler properties
 
 set(COMMON_TARGET_PROPERTIES 
-  POSITION_INDEPENDENT_CODE ON   # -fPIC
   C_STANDARD 99                  # -std=gnu99
   )
 
+if (BUILD_SHARED_LIBS)
+  list(APPEND COMMON_TARGET_PROPERTIES POSITION_INDEPENDENT_CODE ON)   # -fPIC
```

Comment:
I believe that `BUILD_SHARED_LIBS` already implies `-fPIC` on targets that support it. I think we should remove this, also in all other places.

Reply:
It seems that's not the case.

Reply:
I think -fPIC is added automatically. This is the behavior on my mac:
```
C02TC1ABHF1T:test frapet01$ cat lib.c
double f(double x) {
  return x + x * x;
}
C02TC1ABHF1T:test frapet01$ cat CMakeLists.txt
project(LIB LANGUAGES C)
add_library(mylib lib.c)
```
I build without `BUILD_SHARED_LIBS`, and I don't see `-fPIC`:
```
C02TC1ABHF1T:test frapet01$ mkdir build
C02TC1ABHF1T:test frapet01$ cd build/
C02TC1ABHF1T:build frapet01$ cmake ../
-- The C compiler identification is AppleClang 9.0.0.9000039
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc
//...
-- Build files have been written to: /Users/frapet01/projects/test/build
C02TC1ABHF1T:build frapet01$ make VERBOSE=1
/...
[ 50%] Building C object CMakeFiles/mylib.dir/lib.c.o
/Library/Developer/CommandLineTools/usr/bin/cc    -o CMakeFiles/mylib.dir/lib.c.o   -c /Users/frapet01/projects/test/lib.c
[100%] Linking C static library libmylib.a
//...
C02TC1ABHF1T:build frapet01$
```

Then I build with `BUILD_SHARED_LIBS`, and as you can see, `-fPIC` is added:
```
C02TC1ABHF1T:test frapet01$ mkdir build-shared
C02TC1ABHF1T:test frapet01$ cd build-shared/
C02TC1ABHF1T:build-shared frapet01$ cmake ../ -DBUILD_SHARED_LIBS=On
-- The C compiler identification is AppleClang 9.0.0.9000039
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc -- works
-- Detecting C compiler ABI info
// ...
-- Generating done
-- Build files have been written to: /Users/frapet01/projects/test/build-shared
C02TC1ABHF1T:build-shared frapet01$ make VERBOSE=1
// ...
[ 50%] Building C object CMakeFiles/mylib.dir/lib.c.o
/Library/Developer/CommandLineTools/usr/bin/cc -Dmylib_EXPORTS  -fPIC   -o CMakeFiles/mylib.dir/lib.c.o   -c /Users/frapet01/projects/test/lib.c
[100%] Linking C shared library libmylib.dylib
```

I think it makes sense, as `-fPIC` is described as the flag needed to build libraries, so cmake is making a good job in adding it automatically when `BUILD_SHARED_LIBS=On`.

Reply:
The current CMakeLists.txt first compiles the source files into object files, and then add them to a shared lib. So, it's like
```
project(LIB LANGUAGES C)
add_library(libobj OBJECT lib.c)
add_library(mylib $<TARGET_OBJECTS:libobj>)
```
In this case, -fPIC is not specified when the source file is compiled.